Core/src/IfcGeometryConverter/ProfileConverter.h

```diff
@@ -701,23 +674,12 @@ namespace OpenInfraPlatform {
 							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcAsymmetricIShapeProfileDef>(i_shape);
 						if(asym_I_profile) {
 							if(asym_I_profile->TopFlangeWidth) {
-								double bTop = asym_I_profile->TopFlangeWidth * length_factor;
-								double tfTop = tf;
-
-								if(asym_I_profile->TopFlangeThickness) {
-									tfTop = asym_I_profile->TopFlangeThickness * length_factor;
-								}
-								double rTop = r;
-								if(asym_I_profile->TopFlangeFilletRadius) {
-									rTop = asym_I_profile->TopFlangeFilletRadius * length_factor;
-								}
-
-								if(rTop != 0) {
-									addArc(outer_loop, rTop, M_PI, -M_PI_2, tw * 0.5 + rTop, h * 0.5 - tfTop - rTop);
-								}
-								else {
-									outer_loop.push_back(carve::geom::VECTOR(tw * 0.5, (h * 0.5 - tfTop)));
-								}
+								double bTop = asym_I_profile->TopFlangeWidth * length_factor;																
+								double tfTop = asym_I_profile->TopFlangeThickness.value_or(tf);
+								double rTop = asym_I_profile->TopFlangeFilletRadius.value_or(r);
+								//Function GetAsymmetricCoordinates
+								GetAsymmetricIShapeCoordinates(outer_loop, rTop, tw, tfTop, h);

```

Comment:
I don't really like this function. It only adds one corner of the I-Shape and not hte whole as its name conveys. Please leave this functionality currently where it is (so, remove the function and shift the if-statement back). The whole handling of the I-Shape will get its own function.

Reply:
Done